lib/tzinfo/timezone.rb

```diff
@@ -418,6 +418,21 @@ def period_for_local(local, dst = Timezone.default_dst)
       end
     end
 
+    def period_for(tm, dst = Timezone.default_dst)
+      tm = tm.to_orig if tm.is_a?(TimeOrDateTime)
+
+      utc = case tm
+      when Time
+        tm.utc
```

Comment:
Wow. TIL. 10 years of Ruby programming and was always sure that Ruby's `Time` is immutable (at least by this kind of methods). Need to do my homework better.